src/main/kotlin/no/nav/emottak/eventmanager/Routing.kt

```
@@ -31,22 +33,27 @@ fun Application.configureNaisRouts(collectorRegistry: PrometheusMeterRegistry, e
-            val toDate = call.request.queryParameters.get("toDate")
-            if (fromDate.isNullOrEmpty()) {
-                log.info("Mangler parameter: from date")
+            val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")
+            val fromDateParam = call.request.queryParameters.get("fromDate")
+            val toDateParam = call.request.queryParameters.get("toDate")
+            if (fromDateParam.isNullOrEmpty()) {
+                log.info("Mangler parameter: fromDate")
                 call.respond(HttpStatusCode.BadRequest)
             }
-            val fom = SimpleDateFormat("yyyy-MM-dd HH:mm").parse(fromDate).toLocalDateTime()
-            if (toDate.isNullOrEmpty()) {
-                log.info("Mangler parameter: to date")
+            val fromDate = LocalDateTime.parse(fromDateParam, formatter)
+                .atZone(ZoneId.of("UTC"))
+                .toInstant()
+            if (toDateParam.isNullOrEmpty()) {
+                log.info("Mangler parameter: toDate")
                 call.respond(HttpStatusCode.BadRequest)
             }
-            val tom = SimpleDateFormat("yyyy-MM-dd HH:mm").parse(toDate).toLocalDateTime()
-            log.info("Henter hendelser fra events endepunktet til ebms ...")
-            val events = eventsService.fetchevents(fom, tom)
-            log.info("Antall hendelser fra endepunktet : ${events.size}")
-            log.info("Henter siste hendelse : ${events.last()}")
+            val toDate = LocalDateTime.parse(toDateParam, formatter)
+                .atZone(ZoneId.of("UTC"))
+                .toInstant()
+            log.debug("Henter hendelser fra events endepunktet...")
+            val events = eventService.fetchEvents(fromDate, toDate)
+            log.debug("Antall hendelser fra endepunktet : ${events.size}")
+            log.debug("Henter siste hendelse : ${events.lastOrNull()}")
             call.respond(events)
```

Dette er bare små-detaljer, men jeg nevner det i tilfelle du vil rydde litt i rekkefølgen på koden. Akkurat nå er variablene deklarert litt tilfeldig, og det bryter litt med prinsippene:

- What belongs together, sticks together
- Fail fast

Konkret:

- fromDate blir parsed før det i det hele tatt er sjekket om toDateParam finnes, noe som gjør at man kjører unødvendig kode hvis den sjekken feiler.

- Det gir bedre lesbarhet og logikkflyt om toDateParam deklareres nærmere sin validering og bruk.

Ikke kritisk i det hele tatt, men det kan gjøre koden litt lettere å følge og vedlikeholde.